src/utils/vvim.cpp

```diff
@@ -3834,12 +3835,26 @@ void VVim::processPasteAction(QList<Token> &p_tokens, bool p_pasteBefore)
 
     QTextCursor cursor = m_editor->textCursor();
     cursor.beginEditBlock();
-    if (reg.isBlock()) {
-        if (p_pasteBefore) {
+
+    // Different from Vim:
+    //  In visual mode, by default vim select the current char, so paste operation will replace
+    //  current char, but here it is strange to follow this specification
+    bool hasSelection = cursor.hasSelection();
+    bool isVisualLine = checkMode(VimMode::VisualLine);
+    if (hasSelection) {
+        int pos = cursor.selectionStart();
+        cursor.removeSelectedText();
+        if (isBlock && !isVisualLine) {
+            insertChangeBlockAfterDeletion(cursor, pos);
+        }
+    }
+
+    if (isBlock) {
```

Comment:
I suppose it will be simpler to handle two cases independently: has selection and no selection. When there is selection, we do not need to consider pasting before or behind. It is just a suggestion. :)